gpkit/constraints/base.py

```diff
@@ -0,0 +1,222 @@
+"Implements Constraint"
+from ..small_classes import HashVector
+from ..keydict import KeySet, KeyDict
+from ..small_scripts import try_str_without
+from ..repr_conventions import _str, _repr, _repr_latex_
+
+
+class Constraint(list):
```

Comment:
Yeah, I'm not a huge fan of the name, though hopefully an end user would never have to see it...